reactive/kubernetes_worker.py

```diff
@@ -1325,3 +1331,19 @@ def get_registry_location():
         registry = ""
 
     return registry
+
+
+@when('coredns.ready', 'kube-control.dns.available')
+def set_cluster_dns():
```

Comment:
The `coredns.ready` and `kube-control.dns.available flags` are persistent - once they are set, they stay set. That means every time any hook runs, the conditions of this handler will be true, and the handler will run. Meaning that every 5 minutes, when the update-status hook runs, kubelet is going to get restarted. That's no good.

You either need to introduce a new flag to the interface that can be cleared once it has been handled, e.g. `coredns.ip-changed`, or you need to use `data_changed` to check for changes to the IP.

Reply:
You're completely right, thats an error from my part. I'll fix that.